src/main/java/bio/terra/workspace/service/datareference/model/BigQueryDatasetReference.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.workspace.service.datareference.model;
+
+import bio.terra.workspace.generated.model.GoogleBigQueryDatasetUid;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.auto.value.AutoValue;
+
+/**
+ * A representation of a reference to a BigQuery dataset.
+ *
+ * <p>The {@Code JsonTypeName} annotation specifies the class name used for serialization (see the
+ * {@Code JsonSubTypes} annotation in {@Code ReferenceObject} for corresponding deserialization). By
+ * using a constant string instead of the actual class name, changing the name of this class will
+ * not break backwards compatibility with existing serialized objects. This string does not need to
+ * match the class name - it only matches for clarity.
+ */
+@AutoValue
+@JsonTypeName("BigQueryDatasetReference")
+public abstract class BigQueryDatasetReference implements ReferenceObject {
+
+  @JsonCreator
+  public static BigQueryDatasetReference create(
```

Comment:
Can we make builders for these (in a separate PR)? I think you can still call the autovalue constructor in `build()`. Ideally they'd give  you that for free like Immutables does.

Reply:
Do we need builders, or is that for consistency with other autovalue classes? I'm not sure what benefit it brings otherwise

Reply:
It's for style, safety, & immutability. Looks like you do get it for free:
```
 static final class Builder extends DataReference.Builder { // ...
 ```
 💯